File: Selenium/Project/Activity7.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

service = FirefoxService(GeckoDriverManager().install())
with webdriver.Firefox(service=service) as driver:
    wait =WebDriverWait(driver,20)
    driver.get("https://alchemy.hguy.co/jobs/")

    driver.find_element(By.XPATH,"//li[@id ='menu-item-26']/a[1]").click()
    wait.until(expected_conditions.element_to_be_clickable((By.CSS_SELECTOR,"a.button")))
    driver.find_element(By.CSS_SELECTOR,"a.button").click()
    wait.until(expected_conditions.visibility_of_element_located((By.ID,"user_login")))
    driver.find_element(By.ID,"user_login").send_keys("root")
    driver.find_element(By.ID,"user_pass").send_keys("pa$$w0rd")
    driver.find_element(By.ID,"wp-submit").click()
    driver.find_element(By.ID,"job_title").send_keys("Software tester-Virtual1")
    driver.find_element(By.ID,"job_description_ifr").send_keys("Software")
    driver.find_element(By.ID,"company_name").send_keys("IBM")
    driver.find_element(By.XPATH,"//input[@value = 'Preview']").click()
    driver.find_element(By.ID,"job_preview_submit_button").click()
    driver.find_element(By.XPATH,"//li[@id ='menu-item-24']/a[1]").click()
    driver.find_element(By.ID,"search_keywords").send_keys("virtual")
    driver.find_element(By.XPATH,"//input[@value = 'Search Jobs']").click()
    wait.until(expected_conditions.visibility_of_element_located((By.XPATH,"//h3[contains(text(),'Software tester-Virtual')]")))
    Actual = driver.find_element(By.XPATH,"//h3[contains(text(),'Software tester-Virtual')]").text
    Expected = "Virtual"
    if Actual.__contains__(Expected):
       print("Job Post Created Successfully")
       driver.quit()

File: Selenium/Project/Activity7.java
package groupProject;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;
import org.testng.Assert;

public class Activity7 {
    public static void main(String[] args) {
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();

        driver.get("https://alchemy.hguy.co/jobs/");
        Assert.assertEquals(driver.findElement(By.xpath("//h1[@class='site-title']/a")).getText(),
                "Alchemy Jobs", "Failed to navigate to Alchemy Jobs portal");

        //Navigate to Post a Job
        driver.findElement(By.xpath("//li/a[text()='Post a Job']")).click();

        //Sign in
        driver.findElement(By.xpath("//div[contains(@class,'sign-in')]/a")).click();
        driver.findElement(By.id("user_login")).sendKeys("root");
        driver.findElement(By.id("user_pass")).sendKeys("pa$$w0rd");
        driver.findElement(By.id("wp-submit")).click();
        Assert.assertTrue(driver.findElement(By.xpath("//a[contains(text(),'Sign out')]")).isDisplayed(),
                "Sign in is not successful");

        //Filling the form
        String jobTitle = "Test Specialist";
        driver.findElement(By.id("job_title")).sendKeys(jobTitle);
        driver.findElement(By.id("job_location")).sendKeys("Hyderabad");
        Select options = new Select(driver.findElement(By.id("job_type")));
        options.selectByVisibleText("Full Time");
        //driver.switchTo().frame(driver.findElement(By.id("job_description_ifr")));
       // Assert.assertTrue(driver.findElement(By.xpath("//html/body[contains(@class,'job_description')]")).isDisplayed());
        WebElement iframeid = driver.findElement(By.xpath("//iframe[@id='job_description_ifr']"));
        iframeid.sendKeys("Test");
        //driver.switchTo().defaultContent();
        driver.findElement(By.name("submit_job")).click();
      //  Assert.assertEquals(driver.findElement(By.xpath("//div[@class='job_listing_preview_title']/h2")).getText(), "Preview");

        //Submit Listing
        driver.findElement(By.id("job_preview_submit_button")).click();
        Assert.assertTrue(driver.findElement(By.xpath("//div[@class='entry-content clear']")).getText().contains("Job listed successfully."));

        //Verify the job listing in Jobs page
        driver.findElement(By.xpath("//li/a[text()='Jobs']")).click();
        driver.findElement(By.id("search_keywords")).sendKeys(jobTitle);
        driver.findElement(By.xpath("//input[@value='Search Jobs']")).click();
        driver.quit();

    }

}
